src/Psr6Store.php

```diff
@@ -427,6 +464,29 @@ private function saveContentDigest(Response $response): void
         }
     }
 
+    private function gzipResponse(Response $response): void
+    {
+        // Do not gzip if already encoded or not supported
+        if ($response->headers->has('Content-Encoding') ||
+            $response instanceof BinaryFileResponse ||
+            !$this->isGzipSupported() ||
+            $this->isResponseGzipped($response)
+        ) {
+            return;
+        }
+
+        $encoded = gzencode((string) $response->getContent(), $this->options['gzip_level']);
```

Comment:
Should we add a check here for the case that `->getContent()` returns `false` like the streamed response does for example? https://github.com/symfony/symfony/blob/fd530913d494e2b4e9a624ba6350371d8e801ca3/src/Symfony/Component/HttpFoundation/StreamedResponse.php#L129